src/drive/web/modules/selection/duck.spec.jsx

```diff
@@ -0,0 +1,82 @@
+import { combineReducers, createStore } from 'redux'
+import reducer, {
+  showSelectionBar,
+  hideSelectionBar,
+  toggleSelectionBar,
+  isSelectionBarVisible,
+  toggleItemSelection,
+  isSelected,
+  getSelectedFiles
+} from './duck.js'
+
+describe('selection store', () => {
+  const setup = () => createStore(combineReducers({ selection: reducer }))
+
+  const item1 = { id: 1, name: 'item 1' }
+  const item2 = { id: 2, name: 'item 2' }
+
+  it('shows and hides the selection bar', () => {
+    const store = setup()
+
+    store.dispatch(showSelectionBar())
+    expect(isSelectionBarVisible(store.getState())).toBe(true)
+    store.dispatch(showSelectionBar())
+    expect(isSelectionBarVisible(store.getState())).toBe(true)
+
+    store.dispatch(hideSelectionBar())
+    expect(isSelectionBarVisible(store.getState())).toBe(false)
+    store.dispatch(hideSelectionBar())
+    expect(isSelectionBarVisible(store.getState())).toBe(false)
+
+    store.dispatch(toggleSelectionBar())
+    expect(isSelectionBarVisible(store.getState())).toBe(true)
+    store.dispatch(toggleSelectionBar())
+    expect(isSelectionBarVisible(store.getState())).toBe(false)
+  })
+
+  it('selects and deselects item', () => {
+    const store = setup()
+
+    // selecting one item
+    store.dispatch(toggleItemSelection(item1, false))
+    expect(isSelected(store.getState(), item1)).toBe(true)
+    expect(isSelected(store.getState(), item2)).toBe(false)
+    expect(isSelectionBarVisible(store.getState())).toBe(true)
+
+    // reselecting the same item does nothing
+    store.dispatch(toggleItemSelection(item1, false))
+    expect(isSelected(store.getState(), item1)).toBe(true)
```

Comment:
Hmm, I'm not agreed with that. If we select an already selected item it should be removed, no? 

OK the second argument seems to be "selected or not". Can we have a variable for that? In order to understand at the first read the code? 

But I still disagree with the comment / text : "reselecting the same item does nothing" is not accurate.